packages/create-cloudflare/src/deploy.ts

```diff
@@ -99,24 +103,42 @@ export const runDeploy = async (ctx: C3Context) => {
 			: []),
 	];
 
-	const result = await runCommand(deployCmd, {
-		silent: true,
+	const outputFile = join(
+		await mkdtemp(join(tmpdir(), "c3-wrangler-deploy-")),
+		"output.json",
+	);
+
+	await runCommand(deployCmd, {
 		cwd: ctx.project.path,
 		env: {
 			CLOUDFLARE_ACCOUNT_ID: ctx.account.id,
 			NODE_ENV: "production",
+			WRANGLER_OUTPUT_FILE_PATH: outputFile,
 		},
 		startText: "Deploying your application",
 		doneText: `${brandColor("deployed")} ${dim(
 			`via \`${quoteShellArgs(baseDeployCmd)}\``,
 		)}`,
 	});
 
-	const deployedUrlRegex = /https:\/\/.+\.(pages|workers)\.dev/;
-	const deployedUrlMatch = result.match(deployedUrlRegex);
-	if (deployedUrlMatch) {
-		ctx.deployment.url = deployedUrlMatch[0];
-	} else {
+	try {
+		const contents = readFile(outputFile);
+
+		const entries = contents
+			.split("\n")
+			.filter(Boolean)
+			.map((entry) => JSON.parse(entry));
+		const url: string | undefined =
+			entries.find((entry) => entry.type === "deploy")?.targets?.[0] ??
+			entries.find((entry) => entry.type === "pages-deploy")?.url;
+		const deployedUrlRegex = /https:\/\/.+\.(pages|workers)\.dev/;
+		const deployedUrlMatch = url?.match(deployedUrlRegex);
+		if (deployedUrlMatch) {
+			ctx.deployment.url = deployedUrlMatch[0];
+		} else {
+			throw new Error("Failed to find deployment url.");
+		}
```

Comment:
The deployed URL should always be workers.dev or pages.dev, since this is the first deploy from a template—the user won't have had a chance to set up a custom domain/route yet